packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -300,17 +319,44 @@ export class NftDetectionController extends BaseController<
    */
   isMainnet = (): boolean => this.config.chainId === ChainId.mainnet;
 
+  isMainnetByNetworkClientId = (networkClient: NetworkClient): boolean => {
+    return networkClient.configuration.chainId === ChainId.mainnet;
+  };
+
+  private getCorrectChainId({
+    chainId,
+    networkClientId,
+  }: {
+    chainId?: Hex;
+    networkClientId?: NetworkClientId;
+  }) {
+    if (networkClientId) {
+      return this.getNetworkClientById(networkClientId).configuration.chainId;
+    } else if (chainId) {
+      return chainId;
+    }
+    return this.config.chainId;
+  }
```

Comment:
I believe I wrote this but I'm not sure why I did it this way. Could be:
```suggestion
  private getCorrectChainId({
    networkClientId,
  }: {
    networkClientId?: NetworkClientId;
  }) {
    if (networkClientId) {
      return this.getNetworkClientById(networkClientId).configuration.chainId;
     }
    return this.config.chainId;
  }
```
right?